packages/workbox-google-analytics/test/sw/index.js

```diff
@@ -0,0 +1,349 @@
+  }
+};
+
+describe(`initialize`, function() {
+  let idbMap = {};
+
+  const stubIdbHelper = () => {
+    sinon.stub(IDBHelper.prototype, 'get').callsFake(function(key) {
+      const dbStore = `${this._name}|${this._storeName}`;
+      idbMap[dbStore] = idbMap[dbStore] || {};
+      return Promise.resolve(idbMap[dbStore][key]);
+    });
+    sinon.stub(IDBHelper.prototype, 'put').callsFake(function(key, value) {
+      const dbStore = `${this._name}|${this._storeName}`;
+      idbMap[dbStore] = idbMap[dbStore] || {};
+      idbMap[dbStore][key] = value;
+      return Promise.resolve();
+    });
+  };
+
+  const restoreIdbHelper = () => {
+    idbMap = {};
+    IDBHelper.prototype.get.restore();
+    IDBHelper.prototype.put.restore();
+  };
+
+  const trackAddedEventListeners = () => {
+    sinon.spy(self, 'addEventListener');
+  };
+
+  const removeAllEventListeners = () => {
+    for (const [evt, listener] of self.addEventListener.args) {
+      self.removeEventListener(evt, listener);
+    }
+    self.addEventListener.restore();
+  };
+
+  beforeEach(function() {
+    trackAddedEventListeners();
+    stubIdbHelper();
+  });
+
+  afterEach(function() {
+    removeAllEventListeners();
+    restoreIdbHelper();
+  });
+
+  it(`should register a handler to cache the analytics.js script`, function() {
+    sinon.spy(NetworkFirst.prototype, 'handle');
+
+    googleAnalytics.initialize();
+
+    self.dispatchEvent(new FetchEvent('fetch', {
+      request: new Request(
+          `https://${constants.URL.HOST}${constants.URL.ANALYTICS_JS_PATH}`, {
+        mode: 'no-cors',
+      }),
+    }));
+
+    expect(NetworkFirst.prototype.handle.calledOnce).to.be.ok;
+
+    NetworkFirst.prototype.handle.restore();
+  });
+
+  it(`should register GET/POST routes for /collect`, function() {
+    sinon.spy(NetworkOnly.prototype, 'handle');
+
+    googleAnalytics.initialize();
+
+    self.dispatchEvent(new FetchEvent('fetch', {
+      request: new Request(`https://${constants.URL.HOST}` +
+          `${constants.URL.COLLECT_PATH}?${PAYLOAD}`, {
+        method: 'GET',
+      }),
+    }));
+
+    expect(NetworkOnly.prototype.handle.calledOnce).to.be.ok;
+
+    self.dispatchEvent(new FetchEvent('fetch', {
+      request: new Request(`https://${constants.URL.HOST}` +
+          `${constants.URL.COLLECT_PATH}`, {
+        method: 'POST',
+        body: PAYLOAD,
+      }),
+    }));
+
+    expect(NetworkOnly.prototype.handle.calledTwice).to.be.ok;
+
+    NetworkOnly.prototype.handle.restore();
+  });
+
+  it(`should not alter successful hits`, async function() {
+    sinon.stub(self, 'fetch');
+
+    googleAnalytics.initialize();
+
+    self.dispatchEvent(new FetchEvent('fetch', {
+      request: new Request(`https://${constants.URL.HOST}` +
+          `${constants.URL.COLLECT_PATH}?${PAYLOAD}`, {
+        method: 'GET',
+      }),
+    }));
+
+    expect(self.fetch.calledOnce).to.be.ok;
+    expect(self.fetch.firstCall.args[0].url).to.equal(`https://` +
```

Comment:
I'm lost as to what you are checking here. You make a request with a url and check the request had that URL.

Reply:
Yeah, I'm making sure the `initialize()` logic doesn't add anything that modifies the request URL in the non-failure case. Basically I'm making sure that the replay logic doesn't inadvertently apply to non-replayed requests.

Reply:
Another comment request please.